### DISCUSSION TITLE
Infinite Scroll with ForEachStore

### DISCUSSION THREAD

**Discussion by toomozoo** [pointfreeco/swift-composable-architecture#1079] [Q&A]
Hi, I'm working on a view that uses infinite scroll and want to display `ProgressView` after the last item.

For example:

```
List {                                                                                                    
    ForEachStore(self.store.scope(state: \.posts, action: Action.post(id:action:))) {
        PostView(store: $0)

        if viewStore.isRequestInFlight && ViewStore($0).post == viewStore.lastPost {
            ProgressView()
        }
    }
}
```

This works fine, but is this a good way to compare values using `ViewStore($0).post == viewStore.lastPost` ? Or should I not use `ForEachStore` in this case?

**Comment by stephencelis** [comment]
@toomozoo Another approach would be to put the progress view after the `ForEachStore` and avoid needing that `lastPost` check:

```swift
List {                                                                                                    
  ForEachStore(self.store.scope(state: \.posts, action: Action.post(id:action:))) {
    PostView(store: $0)
  }

  if viewStore.isRequestInFlight {
    ProgressView()
  } else {
    // You could also kick off the request here:
    EmptyView().onAppear { viewStore.send(.scrolledToBottom) }
  }
}
```

**Reply by toomozoo** [reply]
@stephencelis
Thank you for your reply! I'm sorry I didn't make it clear enough.
`ProgressView` may appear in the middle of the posts or at the end, so I need to check each of posts.

| Middle of the posts | End  of the posts |
|---|---|
| ![image](https://user-images.githubusercontent.com/104886270/166679040-fa391f6e-621b-40f5-a9aa-9576233b7b08.png) | ![image](https://user-images.githubusercontent.com/104886270/166678417-0f1918d4-9875-4ba1-85d5-1d4be486eb9f.png) |
